src/api-transition-helpers.coffee

```diff
@@ -0,0 +1,17 @@
+LineNumberGutterName = 'line-numbers'
+
+# Given decorationParams/Properties from the 1.0 API, applies changes to enable
+# the 2.0 API under the covers. In 2.0, the `trueType` property will become `type`.
+translateDecorationParamsOldToNew = (decorationParams) ->
```

Comment:
I think we should just automatically fill out any decoration with type `line-number` to have a `gutterName` of `line-numbers`, then adjust the definition of `isType` to return `true` for decorations with `type: line-number` when `gutter` is passed in. We should also automatically insert a `gutterName` of `line-numbers` for `gutter` decorations without a `gutterName` for now (but still have a deprecation warning for this case). I'm not sure we need to break this out into a separate helper.